packages/react/src/hooks/useLauncher.tsx

```diff
@@ -0,0 +1,84 @@
+import {
+  getTreasureLauncherAuthToken,
+  startUserSessionViaLauncher,
+} from "@treasure-dev/launcher";
+import type { SessionOptions, TDKAPI, User } from "@treasure-dev/tdk-core";
+import { type ReactNode, useCallback, useEffect } from "react";
+import { AccountModal } from "../components/launcher/AccountModal";
+import { setStoredAuthToken } from "../utils/store";
+
+type Props = {
+  getAuthTokenOverride?: () => string | undefined;
+  tdk: TDKAPI;
+  setRootElement: (el: ReactNode) => void;
+  setUser: (user: User) => void;
+  onConnect?: (user: User) => void;
+};
+
+export const useLauncher = ({
+  getAuthTokenOverride,
+  tdk,
+  setUser,
+  setRootElement,
+  onConnect,
+}: Props) => {
+  const getAuthToken = useCallback(() => {
+    return getAuthTokenOverride?.() ?? getTreasureLauncherAuthToken();
+  }, [getAuthTokenOverride]);
+
+  const isUsingTreasureLauncher = useCallback((): boolean => {
+    return getAuthToken() !== undefined;
+  }, [getAuthToken]);
```

Comment:
Will the value of this change during runtime? Can we pass a `boolean` result through instead of a function?